src/server/system_services/bucket_server.js

```diff
@@ -1284,9 +1284,16 @@ function validate_bucket_creation(req) {
         !VALID_BUCKET_NAME_REGEXP.test(req.rpc_params.name.unwrap())) {
         throw new RpcError('INVALID_BUCKET_NAME');
     }
-    if (req.system.buckets_by_name && req.system.buckets_by_name[req.rpc_params.name.unwrap()]) {
-        throw new RpcError('BUCKET_ALREADY_EXISTS');
+    var bucket = req.system.buckets_by_name && req.system.buckets_by_name[req.rpc_params.name.unwrap()];
+
+    if (bucket) {
+        if (bucket.owner_account._id === req.account._id) {
```

Comment:
This comparison does not work for ObjectId type.
Need to compare the string value of it or use the `system_store.has_same_id()` function.

Reply:
have used system_store.has_same_id()